### PR TITLE
.NET SDK for Microsoft.Workloads/SapDiscoverySites api version 2023-10-01-preview

### PR DESCRIPTION
**Pull request by gargankit-microsoft**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

Namespace approval: https://github.com/Azure/azure-sdk-pr/issues/1737

Target date for PR approvals for PLR approvals: 10/Mar/2024

### CODE REVIEW

**Comment by gargankit-microsoft** [comment]
Hi @ArthurMa1978 

We have an unusual scenario in one of our operations: Microsoft.Workloads/SapDiscoverySites POST ImportEntities for which we need your help in testing the generated SDK. [We have shared the same query in support channel as well](https://teams.microsoft.com/l/message/19:7b87fb348f224b37b6206fa9d89a105b@thread.skype/1709288622409?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1709288622409&teamName=Azure%20SDK&channelName=Language%20-%20DotNet&createdTime=1709288622409)

The user needs to monitor the operation status for various internal states in order to carry out operations. Therefore, there must be a method to:

Retrieve OperationStatus on demand after making this post call and getting Azure-Async-Operation header.
Since properties are not inherently modeled, we require another model than default that inherits from the existing OperationStatus model to access these properties for OperationStatus. The property Payload includes the following fields (all optional):
```
{
	"discoveryExcelSasUri": <string>,
	"errorExcelSasUri": <string>,
	"status": <ENUM Accepted|AwaitingFile|ProcessingFile|Succeeded|UploadingErrorExcel|PartiallySucceeded|Failed|Canceled>,
	"rowsImported": <int>,
	"totalRows": <int>
}
```

**Comment by ArthurMa1978** [comment]
> Hi @ArthurMa1978 Arthur Ma FTE
> 
> We have an unusual scenario in one of our operations: Microsoft.Workloads/SapDiscoverySites POST ImportEntities for which we need your help in testing the generated SDK. [We have shared the same query in support channel as well](https://teams.microsoft.com/l/message/19:7b87fb348f224b37b6206fa9d89a105b@thread.skype/1709288622409?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1709288622409&teamName=Azure%20SDK&channelName=Language%20-%20DotNet&createdTime=1709288622409)
> 
> The user needs to monitor the operation status for various internal states in order to carry out operations. Therefore, there must be a method to:
> 
> Retrieve OperationStatus on demand after making this post call and getting Azure-Async-Operation header. Since properties are not inherently modeled, we require another model than default that inherits from the existing OperationStatus model to access these properties for OperationStatus. The property Payload includes the following fields (all optional):
> 
> ```
> {
> 	"discoveryExcelSasUri": <string>,
> 	"errorExcelSasUri": <string>,
> 	"status": <ENUM Accepted|AwaitingFile|ProcessingFile|Succeeded|UploadingErrorExcel|PartiallySucceeded|Failed|Canceled>,
> 	"rowsImported": <int>,
> 	"totalRows": <int>
> }
> ```

The test code can be this:
```
public virtual async Task test()
        {
            var operation = await ImportEntitiesAsync(WaitUntil.Started).ConfigureAwait(false);
            while (operation.HasCompleted)
            {
                Response statusResponse = await operation.UpdateStatusAsync().ConfigureAwait(false);
                if(statusResponse.Status == 202)
                {
                    // deserialize the statusResponse.Content and check status)
                }
                else
                {
                    // exception
                }
            }
        }
```

File: sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Microsoft Azure MigrationDiscoverySap management client library for .NET
+
+**[Describe the service briefly first.]**
```
Comment (ArthurMa1978) [inline sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/README.md:3]: Update the description of the service here.

**Comment by gargankit-microsoft** [comment]
Hi @ArthurMa1978 , 
Please help with below queries

- The process to add customization code for a new class for operation status specific to our use case and enums and custom properties for users to parse response into.
- we are unable to push recording to assets repo as we dont have access to write to assets repo, we followed steps in this link, [Request access to Azure REST API and SDK repositories (eng.ms)](https://eng.ms/docs/products/azure-developer-experience/onboard/access?tabs=write-access), but didnt got any way to get access for assets repo.

**Comment by gargankit-microsoft** [comment]
@ArthurMa1978 
all changes are checked in, Please review the PR.

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**